addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -22,6 +26,101 @@ def setUpClass(cls):
         super(TestMassMailValues, cls).setUpClass()
         cls._create_mailing_list()
 
+    @users('user_marketing')
+    def test_mailing_body_cropped_vml_image(self):
+        """ Testing mail mailing responsive bg-image cropping for Outlook.
+
+        Outlook needs background images to be converted to VML but there is no
+        way to emulate `background-size: cover` that works for Windows Mail as
+        well. We therefore need to crop the image in the VML version to mimick
+        the style of other email clients.
+        """
+        attachment = {}
+        def patched_get_image(self, url, session):
+            return base64.b64decode(BASE_64_STRING)
+        original_images_to_urls = self.env['mailing.mailing']._create_attachments_from_inline_images
+        def patched_images_to_urls(self, b64images):
+            urls = original_images_to_urls(b64images)
+            if len(urls) == 1:
+                (attachment_id, attachment_token) = re.search(r'/web/image/(?P<id>[0-9]+)\?access_token=(?P<token>.*)', urls[0]).groups()
+                attachment['id'] = attachment_id
+                attachment['token'] = attachment_token
+                return urls
+            else:
+                return []
+        with patch("odoo.addons.mass_mailing.models.mailing.MassMailing._get_image_by_url",
+                   new=patched_get_image), \
+             patch("odoo.addons.mass_mailing.models.mailing.MassMailing._create_attachments_from_inline_images",
+                   new=patched_images_to_urls):
+            mailing = self.env['mailing.mailing'].create({
+                'name': 'Test',
+                'subject': 'Test',
+                'state': 'draft',
+                'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+                'body_html': """
+                    <html>
+                        <!--[if mso]>
+                            <v:image src="https://www.example.com/image" style="width:100px;height:100px;"/>
+                        <![endif]-->
+                    </html>
+                """,
+            })
+        self.assertEqual(str(mailing.body_html), """
```

Comment:
Could probably use fstrings instead of .format ? More inlined with current code :)